src/lib/RelayOrderLib.sol

```diff
@@ -1,43 +1,76 @@
+struct RelayInput {
+    // The ERC20 token address
+    ERC20 token;
+    // The time at which the input start decaying
+    uint256 decayStartTime;
+    // The time at which the price becomes static
+    uint256 decayEndTime;
+    // The amount of tokens at the start of the time period
+    uint256 startAmount;
+    // The amount of tokens at the end of the time period
+    uint256 endAmount;
+    // The address who must receive the tokens to satisfy the order
+    address recipient;
+}
+
+/// @dev An amount of output tokens that decreases linearly over time
+struct RelayOutput {
+    // The ERC20 token address (or native ETH address)
+    address token;
+    // The time at which the output start decaying
+    uint256 decayStartTime;
+    // The time at which the price becomes static
+    uint256 decayEndTime;
+    // The amount of tokens at the start of the time period
+    uint256 startAmount;
+    // The amount of tokens at the end of the time period
+    uint256 endAmount;
+    // The address who must receive the tokens to satisfy the order
+    address recipient;
+}
 
 /// @dev External struct used to specify simple relay orders
 struct RelayOrder {
     // generic order information
     OrderInfo info;
-    // The time at which the inputs start decaying
-    uint256 decayStartTime;
-    // The time at which price becomes static
-    uint256 decayEndTime;
     // ecnoded actions to execute onchain
     bytes[] actions;
     // The tokens that the swapper will provide when settling the order
-    InputTokenWithRecipient[] inputs;
+    RelayInput[] inputs;
+    // The tokens that must be received to satisfy the order
+    RelayOutput[] outputs;
 }
 
 /// @notice helpers for handling relay order objects
 library RelayOrderLib {
     using OrderInfoLib for OrderInfo;
 
     bytes private constant INPUT_TOKEN_TYPE =
-        "InputTokenWithRecipient(address token,uint256 amount,uint256 maxAmount,address recipient)";
+        "RelayInput(address token,uint256 decayStartTime,uint256 decayEndTime,uint256 startAmount,uint256 endAmount,address recipient)";
+
+    bytes private constant OUTPUT_TOKEN_TYPE =
+        "RelayOutput(address token,uint256 decayStartTime,uint256 decayEndTime,uint256 startAmount,uint256 endAmount,address recipient)";
 
     bytes32 private constant INPUT_TOKEN_TYPE_HASH = keccak256(INPUT_TOKEN_TYPE);
+    bytes32 private constant OUTPUT_TOKEN_TYPE_HASH = keccak256(OUTPUT_TOKEN_TYPE);
 
     bytes internal constant ORDER_TYPE = abi.encodePacked(
         "RelayOrder(",
         "OrderInfo info,",
-        "uint256 decayStartTime,",
-        "uint256 decayEndTime,",
         "bytes[] actions,",
-        "InputTokenWithRecipient[] inputs,",
-        "OutputToken[] outputs)",
+        "RelayInput[] inputs,",
+        "RelayOutput[] outputs)",
+        INPUT_TOKEN_TYPE,
         OrderInfoLib.ORDER_INFO_TYPE,
-        INPUT_TOKEN_TYPE
+        OUTPUT_TOKEN_TYPE_HASH
```

Comment:
this should be OUTPUT_TOKEN_TYPE

Reply:
note to add a test suite so we can check the proper 712 spec... that is a larger task but I'll add an issue cause its easy to run into this stuff w witness 👀

Reply:
Ah yes this is fixed in #28